GeofenceDisplay/src/geofence-display.ts

```diff
@@ -2,153 +2,82 @@ export {}
+import './rastercoords.js'
 
 const PlantquestGeofenceDisplay = L.Layer.extend({
-  options: {
-    // geofenceLabelPane.style.zIndex = 235
-    // provide geofence data to component using an option
-  },
-
   initialize: function (this: any, options: any) {
-    console.log('PGeoD init')
+    console.log('GeofenceDisplay init function called.')
+    L.Util.setOptions(this, options)
     this._state = {
-      // this._state.GeofenceByID
-      // self.geofence.map[ent.id] = self.config.geofence.prepare(geofence) || geofence
-      //
-      // self.data goes in here
-      // self.data.geofence = ['mock-data-1', 'mock-data-2', 'mock-data-3']
+      zindex: 0,
+      geofenceByID: {},
     }
-    L.Util.setOptions(this, options)
   },
 
   onAdd: function (this: any, _map: any) {
-    console.log('PGeoD onAdd', this.options)
-    // let div = L.DomUtil.create('div')
-
-    let geo = new Geofence(null, null)
-    console.log('geo:', geo)
-
-    geo.show(_map)
-
-    // // TODO: Define format and value of mock geofence data in dev/main.js
-    // self.data.geofence = ['mock-data-1', 'mock-data-2', 'mock-data-3']
-
-    // // TODO: Extract geofence.prepare() function
-    // self.data.geofence.forEach((ent) => {
-    //   let geofence = new Geofence(ent, ctx)
-    //   self.geofence.map[ent.id] =
-    //     self.config.geofence.prepare(geofence) || geofence
-    // })
-
-    // // GeofencePane related code
-    // // TODO: Tie in with existing setup
-    // _map.createPane('geofence')
-    // let geofencePane = _map.getPane('geofence')
-    // geofencePane.style.zIndex = 230
-    // geofencePane.style.pointerEvents = 'none'
-
-    // _map.createPane('geofenceLabel')
-    // let geofenceLabelPane = _map.getPane('geofenceLabel')
-    // geofenceLabelPane.style.zIndex = 235
-    // geofenceLabelPane.style.pointerEvents = 'none'
-
-    // self.layer.geofence = L.layerGroup(null, { pane: 'geofence' }).addTo(_map)
-    // self.layer.geofence.name$ = 'geofence'
-
-    // //
-
-    // // TODO: show/clear geofence functionality required here?
-    // self.showGeofence = function (geofence, show) {
-    //   if (null == geofence) {
-    //     return
-    //   }
-
-    //   show = !!show
-
-    //   if (true === show) {
-    //     geofence.show(self.layer.geofence)
-    //   } else if (false === show) {
-    //     geofence.hide()
-    //   }
-    // }
-
-    // self.clearGeofences = function () {
-    //   for (let geofenceID in self.geofence.map) {
-    //     let geofence = self.geofence.map[geofenceID]
-    //     delete self.geofence.map[geofenceID]
-    //     if (geofence && geofence.hide) {
-    //       geofence.hide()
-    //     }
-    //   }
-    // }
-
-    // // TODO: add geofencePane to div
-
-    return _map
+    console.log('GeofenceDisplay onAdd function called.')
+    let self = this
+    console.log('Local geofence variable before add:', self._state.geofenceByID)
+    for (let geodata in self.options.geofences) {
```

Comment:
geodata is a string, thus needs a better name - "data" would generally mean a complex type
also, this is an iteration over an array, so the `in` operator is not appropriate - it is more normally used for iteration over the keys of a map
instead use the traditional indexed form

Reply:
also, why use code that is different from the original?
```
    // self.data.geofence.forEach((ent) => {
    //   let geofence = new Geofence(ent, ctx)
    //   self.geofence.map[ent.id] =
    //     self.config.geofence.prepare(geofence) || geofence
    // })
```

And in particular, the `prepare` method is now not being called

Reply:
`geodata` variable changed to `fence`, and `for x in y` changed to iterator more suited to arrays

Reply:
Concerning the code, it was moved as you requested into a `geofenceByID` object in `_state`.

As for the prepare method, according to pqam.js all it did was the following: `(x)=>x`, so I left it out. Should it be reintroduced as a separate function?